lib/_io.scm

```diff
@@ -8343,6 +8343,167 @@
         (macro-dynamic-bind output-port port thunk)))))
 
 ;;;----------------------------------------------------------------------------
+(define-prim (##open-raw-device
+              direction
+              name
+              fd)
```

Comment:
Right, that function already exists and constructs the port. 
I will remove this function, per your comment about caller doing the opening of the device, as this function doesn't do much otherwise.